pasqal_cloud/endpoints.py

```diff
@@ -16,29 +16,93 @@
 
 from dataclasses import dataclass
 
+import typing
+
 if version_info[:2] >= (3, 8):
     from typing import Final
 else:
     from typing_extensions import Final  # type: ignore
 
 
-CORE_API_URL: Final[str] = "https://apis.pasqal.cloud/core-fast"
+# ---- Endpoints ----
+
+PROD_CORE_API_URL: Final[str] = "https://apis.pasqal.cloud/core-fast"
+PREPROD_CORE_API_URL: Final[str] = "https://apis.preprod.pasqal.cloud/core-fast"
+DEV_CORE_API_URL: Final[str] = "https://apis.dev.pasqal.cloud/core-fast"
 
 
 @dataclass
 class Endpoints:
-    core: str = CORE_API_URL
+    core: str = PROD_CORE_API_URL
+
+
+# ---- Auth0 ----
+
+PROD_AUTH0_DOMAIN: Final[str] = "pasqal.eu.auth0.com"
+PROD_PUBLIC_CLIENT_ID: Final[str] = "PeZvo7Atx7IVv3iel59asJSb4Ig7vuSB"
+PROD_AUDIENCE: Final[str] = "https://apis.pasqal.cloud/account/api/v1"
+
+PREPROD_AUTH0_DOMAIN: Final[str] = "pasqal-preprod.eu.auth0.com"
+PREPROD_PUBLIC_CLIENT_ID: Final[str] = "2l6A2ldvwJE5sdkghu40BTYLm7sSUAv9"
+PREPROD_AUDIENCE: Final[str] = "https://apis.preprod.pasqal.cloud/account/api/v1"
 
+DEV_AUTH0_DOMAIN: Final[str] = "pasqal-dev.eu.auth0.com"
+DEV_PUBLIC_CLIENT_ID: Final[str] = "5QtfSu1UV118Iz6By6IJRSNoDrLbAiOv"
+DEV_AUDIENCE: Final[str] = "https://apis.dev.pasqal.cloud/account/api/v1"
 
-AUTH0_DOMAIN: Final[str] = "pasqal.eu.auth0.com"
-PUBLIC_CLIENT_ID: Final[str] = "PeZvo7Atx7IVv3iel59asJSb4Ig7vuSB"
-AUDIENCE: Final[str] = "https://apis.pasqal.cloud/account/api/v1"
 REALM: Final[str] = "pcs-users"
 
 
 @dataclass
 class Auth0Conf:
-    domain: str = AUTH0_DOMAIN
-    public_client_id: str = PUBLIC_CLIENT_ID
-    audience: str = AUDIENCE
+    domain: str = PROD_AUTH0_DOMAIN
+    public_client_id: str = PROD_PUBLIC_CLIENT_ID
+    audience: str = PROD_AUDIENCE
     realm: str = REALM
+
+
+# ---- Environment ----
+
+ENV_OPTIONS = ("prod", "preprod", "dev")
+
+
+def init_endpoints(env: str) -> typing.Tuple[Endpoints, Auth0Conf]:
+    if env not in ENV_OPTIONS:
+        raise ValueError("This environment does not exist, valid options are {}".format(', '.join(ENV_OPTIONS)))
```

Comment:
Can you use f-strings instead of format? I don't think we're locked on a version of python that doesn't support right?